plugins/src/opengrok/auth/plugin/LdapUserPlugin.java

```diff
@@ -125,22 +146,19 @@ public void fillSession(HttpServletRequest req, User user) {
             return;
         }
 
-        if (!records.containsKey("uid") || records.get("uid").isEmpty()) {
-            LOGGER.log(Level.WARNING, "uid record for user {0} is not present or empty",
-                    user);
-            return;
+        for (String attrName : attributes) {
+            if (!records.containsKey(attrName) || records.get(attrName).isEmpty()) {
+                LOGGER.log(Level.WARNING, "{0} record for user {1} is not present or empty",
+                        new Object[]{attrName, user});
+            }
         }
 
-        if (!records.containsKey("mail") || records.get("mail").isEmpty()) {
-            LOGGER.log(Level.WARNING, "mail record for user {0} is not present or empty",
-                    user);
-            return;
+        Map<String, Set<String>> attrSet = new HashMap<>();
+        for (String attrName : attributes) {
+            attrSet.put(attrName, records.get(attrName));
```

Comment:
you now copy the map to a new map, right? isn't copy constructor better solution?

Reply:
`attributes` is array

Reply:
Right. I didn't see that.